src/main/scala/uk/gov/nationalarchives/tdr/api/service/ConsignmentStatusService.scala

```diff
@@ -0,0 +1,21 @@
+package uk.gov.nationalarchives.tdr.api.service
+
+import uk.gov
+import uk.gov.nationalarchives
+import uk.gov.nationalarchives.Tables
+import uk.gov.nationalarchives.Tables.ConsignmentstatusRow
+import uk.gov.nationalarchives.tdr.api.db.repository.ConsignmentStatusRepository
+import uk.gov.nationalarchives.tdr.api.graphql.fields.ConsignmentFields.CurrentStatus
+
+import java.util.UUID
+import scala.concurrent.{ExecutionContext, Future}
+
+class ConsignmentStatusService(consignmentStatusRepository: ConsignmentStatusRepository)
+                              (implicit val executionContext: ExecutionContext) {
+
+  def getConsignmentStatus(consignmentId: UUID): Future[CurrentStatus] = {
+    for {
+      upload <- consignmentStatusRepository.getConsignmentStatus(consignmentId).map(_.map(uploadRow => uploadRow.value).headOption)
```

Comment:
I don't think we can rely on `headOption` always getting the most recent status. This is actually a complicated question, though - should we get the most recent status by date? Or does "UploadComplete" always beat "UploadStarted"?

We might need to have a chat about this to figure out the data model, even if it's a preliminary one while this is the only status.

Reply:
Yeah, this one is something I'm really not 100% sure about, what I've done only really works with the single case we have for the consignment status at the moment and wouldn't work as we expand its use. I think sorting by date could be useful since then we know the most recent progress, but that's probably more applicable to when we're using consignment status to check stages other than the upload. Whereas the "UploadComplete" would work for what we're doing at the moment.

I do like the idea of getting the most recent status row though since we would know that if the most recent one is "UploadComplete" we don't need to check if they've started an upload or filled in the transfer agreement form etc. But I agree it's worth having a chat about so we can make sure we're on the same page 👌

Reply:
We had a chat about this. Most recent status makes sense so I've changed the repository method to order by created date descending and then `headOption` should return the newest.
I've added in a repository test to check that the ordering is working.

Reply:
Ah, I was forgetting that the business logic was in the service layer, sorry. With `.headOption`, it's having to rely on the repository ordering by date. It seems like it might be a bit fragile, because the overall logic only works when you put them together. If you move the logic into the service layer, you can check it with a unit test.

Reply:
It's a fair point. I've removed the sorting from the repository and added it into the service method and I've added a unit test for checking the sorting.